secant/Features/Settings/UIKitBridge/UIShareDialog.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  UIShareDialog.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 30.01.2023.
+//
+
+import Foundation
+import UIKit
+import SwiftUI
+
+public class UIShareDialog: UIView {
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+}
+
+extension UIShareDialog {
+    public func doInitialSetup(tempSDKDir: URL, tempWalletDir: URL, tempTCADir: URL, completion: @escaping () -> Void) {
+        DispatchQueue.main.async {
+            let activityVC = UIActivityViewController(activityItems: [tempSDKDir, tempWalletDir, tempTCADir], applicationActivities: nil)
+            
+            UIApplication.shared.connectedScenes.map({ $0 as? UIWindowScene })
+            .compactMap({ $0 })
+            .first?.windows.first?.rootViewController?.present(
+                activityVC,
+                animated: true,
+                completion: completion
+            )
+        }
+    }
+}
+
+struct UIShareDialogView: UIViewRepresentable {
+    let tempSDKDir: URL
+    let tempWalletDir: URL
+    let tempTCADir: URL
+    let completion: () -> Void
+
+    public func makeUIView(context: UIViewRepresentableContext<UIShareDialogView>) -> UIShareDialog {
+        let view = UIShareDialog()
+        view.doInitialSetup(tempSDKDir: tempSDKDir, tempWalletDir: tempWalletDir, tempTCADir: tempTCADir, completion: completion)
+        return view
+    }
+    
+    public func updateUIView(_ uiView: UIShareDialog, context: UIViewRepresentableContext<UIShareDialogView>) { }
```

Comment:
Please debug the information passed through `updateUIView`. It's often showcased as a method without a body, but it's not correct.  this method is invoked on SwiftUI re-renders to informed the UIViews of the new updates.

reference: https://medium.com/nerd-for-tech/introducing-lottie-animations-to-swiftui-7f298e324c23

Reply:
After a few attempts and investigations we can be sure it's not going to bite us down the road to leave the method empty. I left the comment in the body so anybody see the reasons why.